frontend/sw360-portlet/src/main/resources/META-INF/resources/html/admin/user/view.jsp

```diff
@@ -114,23 +128,112 @@
+                addRowToSecGrpsAndRolesTable();
+            });
+            $('#secGroupsAndRolesTable').on('click', 'svg[data-row-id]', function(event) {
+                var rowId = $(event.currentTarget).data().rowId;
+                $('#' + rowId).remove();
+            });
+
+            $("#secGrpsUserForm").submit(function(event) {
+                event.preventDefault();
+                let formData = {}, formDataText = {};
+                $("#secGroupsAndRolesTable tr").each(function() {
+                    let groupElement = $(this).find("td:first input");
+                    if(!groupElement.length) {
+                        return;
+                    }
+                    let group = groupElement.val();
+                    let role = $(this).find("td:eq(1) select").val(),
+                        roles = formData[group],
+                        roleText = $(this).find("td:eq(1) select option:selected").text(),
+                        rolesText = formDataText[group];
+                    if(!roles) {
+                        roles = new Array();
+                        rolesText = new Array();
+                    }
+                    if(roles.indexOf(role) == -1) {
+                        roles.push(role);
+                        rolesText.push(roleText);
+                    }
+
+                    formData[group] = roles;
+                    formDataText[group] = rolesText;
+                });
+                let emailOfUser = $("#emailOfUser").val();
+                $.ajax({
+                    type: 'POST',
+                    url : '<%= editsecgroupurl %>',
+                    data: JSON.stringify({"formData" : formData, "email" : emailOfUser}),
+                    accept: 'application/json',
+                    contentType: 'application/json',
+                    cache: false,
+                    success: function(data) {
+                        updateAlertMsg('<liferay-ui:message key="SUCCESS" /> !', '<liferay-ui:message key="secondary.departments.and.roles.edited.successfully.for.user" /> : ' + emailOfUser, "alert alert-success")
+                        updateSecGrpRolesTD(formDataText);
+                    },
+                    error: function(data) {
+                        updateAlertMsg('<liferay-ui:message key="error" /> !', '<liferay-ui:message key="failed.to.edit.secondary.departments.and.roles.edited.for.user" /> : ' + emailOfUser, "alert alert-warning")
+                    }
+                });
+            });
 
+            var secGrpsUserFormDiv = $("#secGrpsUserFormDiv").clone(true, true);
+            secGrpsUserFormDiv.removeClass("d-none");
+            $("#secGrpsUserFormDiv").remove();
+            $(".editUser").click(function(){
+                dialog.confirm('info', 'users', '<liferay-ui:message key="edit.users.secondary.departments.and.role" />', secGrpsUserFormDiv.clone(true, true), '<liferay-ui:message key="edit" />', {
```

Comment:
Button with label `Save` would make more sense instead of `Edit` in modal pop-up.
& how about keeping 1 row added y default instead of making user add the very first row?

![image](https://user-images.githubusercontent.com/50870174/116873144-c945be00-ac34-11eb-89a5-8a41e669831e.png)

One more thing, clicking on `Edit` button without adding any rows, makes a call to backend and show the success/error message, as shown in above screenshot. (not an appropriate use case, ideally `Save` button should be disabled or a backend call should not happen unless a row is added)

Reply:
>I guess button with label Save would make more sense here instead of Edit in modal pop-up.

   - Changed Edit button to `Save` button

>One more thing, clicking on Edit button without adding any rows, makes a call to backend and show the success/error message, as shown in above screenshot. (not an appropriate use case, ideally Save button should be disabled or a backend call should not happen unless a row is added)

Use case - Save without any row
 - User has some Secondary roles and Groups, now all of those secondary group and roles needs to be removed.

>  how about keeping 1 row added by default on load of modal pop-up instead of making user add the very first row?
- kindly refer above use-case, 
- Since it is optional to have secondary roles , it was made inline with other parts of sw360 like `external id` and `additional roles`
![image](https://user-images.githubusercontent.com/56534710/117108156-6a9d5300-ada0-11eb-8f53-0bc8f01b8595.png)